src/server/scripts/EasternKingdoms/zone_westfall.cpp

```diff
@@ -169,11 +170,361 @@ class spell_westfall_quest_credit_jangolode_event : public SpellScript
+        {
+           case OPTION_QUESTION:
+           {
+               clueGainChance = 25;
+               break;
+           }
+           case OPTION_PAY:
+           {
+               clueGainChance = 75;
+               bribeFailed = true;
+               break;
+           }
+           default:
+               return true;
+        }
+
+        me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+        CloseGossipMenuFor(player);
+        if (roll_chance_i(clueGainChance))
+            GiveClue(player);
+        else
+        {
+            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+            AttackStart(player);
+        }
+
+        return false;
+    }
+
+    void GiveClue(Player* player)
+    {
+        if (!player)
+            return;
+
+        _events.CancelEventGroup(EVENT_GROUP_OOC);
+        me->SetFacingToObject(player);
+
+        uint16 slot = player->FindQuestSlot(MURDER_WAS_THE_CASE_THAT_THEY_GAVE_ME);
+        if (player->GetQuestSlotCounter(slot, CLUE1) == 0)
+        {
+            player->CastSpell(player, SPELL_HOBO_INFORMATION_1);
+            Talk(SAY_CLUE1, player);
+            me->DespawnOrUnsummon(12s);
+            return;
+        }
+        if (player->GetQuestSlotCounter(slot, CLUE2) == 0)
+        {
+            player->CastSpell(player, SPELL_HOBO_INFORMATION_2);
+            Talk(SAY_CLUE2, player);
+            me->DespawnOrUnsummon(12s);
+            return;
+        }
+        if (player->GetQuestSlotCounter(slot, CLUE3) == 0)
+        {
+            player->CastSpell(player, SPELL_HOBO_INFORMATION_3);
+            Talk(SAY_CLUE3, player);
+            me->DespawnOrUnsummon(12s);
+            return;
+        }
+        if (player->GetQuestSlotCounter(slot, CLUE4) == 0)
+        {
+            player->CastSpell(player, SPELL_HOBO_INFORMATION_4);
+            Talk(SAY_CLUE4, player);
+            me->DespawnOrUnsummon(12s);
+            return;
+        }
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        me->SetAIAnimKitId(0);
+        _events.CancelEventGroup(EVENT_GROUP_OOC);
+
+        if (!who->IsPlayer())
+            return;
+
+        if (bribeFailed)
+            Talk(SAY_AGGRO_BRIBE, who);
+        else
+            Talk(SAY_AGGRO_CONVINCE, who);
+    }
+
+    void DamageTaken(Unit* attacker, uint32& damage) override
+    {
+        if (!flee && me->HealthBelowPctDamaged(20, damage))
+        {
+            flee = true;
+            me->DoFleeToGetAssistance();
+            Talk(SAY_FLEE);
+        }
+    }
+
+    void JustDied(Unit* who) override
+    {
+        hoboRage = false;
+        ENSURE_AI(npc_westfall_hobo_witness, who->ToCreature()->AI())->hoboRage = false;
+        me->CastSpell(me, SPELL_SUMMON_RAGAMUFFIN_LOOTER);
+    }
+
+    void EnterEvadeMode(EvadeReason why) override
+    {
+        ScriptedAI::EnterEvadeMode(why);
+    }
+
+    void Reset() override
+    {
+        ScriptedAI::Reset();
+        hoboRage = false;
```

Comment:
_events.Reset() is missing. You have to reset the eventmap otherwise the events will keep happening after a reset